src/test/ui/asm/x86_64/bad-reg.rs

```diff
@@ -29,8 +29,6 @@ fn main() {
         //~^ ERROR invalid register `rsp`: the stack pointer cannot be used as an operand
         asm!("", in("ip") foo);
         //~^ ERROR invalid register `ip`: the instruction pointer cannot be used as an operand
-        asm!("", in("k0") foo);
```

Comment:
Why is this being removed? It is still invalid to use `k0` as an input/output operand, right?

Reply:
No, it's now valid. I previously incorrectly thought `k0` wasn't a real register, like `xzr` on AArch64.

Reply:
I think in that case the description should be updated

> However k0 is a valid register and we need to have a way of marking it as clobbered for clobber_abi.

somewhat misleadingly led me to believe that `k0` was only being exposed for use as a clobber.